samples/nrf_desktop/src/modules/power_manager.c

```diff
@@ -241,41 +245,42 @@ static bool event_handler(const struct event_header *eh)
 			system_off();
 		}
 
+		power_down_counter_reset();
+
 		return false;
 	}
 
-	if (IS_ENABLED(CONFIG_DESKTOP_USB_ENABLE)) {
-		if (is_usb_state_event(eh)) {
-			const struct usb_state_event *event =
-				cast_usb_state_event(eh);
-
-			switch (event->state) {
-			case USB_STATE_POWERED:
-				usb_connected = true;
-				k_delayed_work_cancel(&power_down_trigger);
-				if (power_state != POWER_STATE_IDLE) {
-					struct wake_up_event *wue =
-						new_wake_up_event();
-					EVENT_SUBMIT(wue);
-				}
-				break;
-
-			case USB_STATE_DISCONNECTED:
-				usb_connected = false;
-				k_delayed_work_submit(&power_down_trigger,
-						      POWER_DOWN_CHECK_MS);
-				break;
-
-			default:
-				/* Ignore */
-				break;
+	if (IS_ENABLED(CONFIG_DESKTOP_USB_ENABLE) && is_usb_state_event(eh)) {
+		const struct usb_state_event *event = cast_usb_state_event(eh);
+
+		switch (event->state) {
+		case USB_STATE_POWERED:
+			usb_connected = true;
+			k_delayed_work_cancel(&power_down_trigger);
+			if (power_state != POWER_STATE_IDLE) {
+				struct wake_up_event *wue =
+					new_wake_up_event();
+				EVENT_SUBMIT(wue);
 			}
-			return false;
+			break;
+
+		case USB_STATE_DISCONNECTED:
+			usb_connected = false;
+			power_down_counter_reset();
```

Comment:
Good idea to reset the counter on USB disconnect.

Reply:
yep

Reply:
nrf_desktop CI is failing with power_manager.c:142 assert. Should not be merged until this is fixed.

Reply:
Looks like a `break` is missing after `POWER_STATE_OFF`. Didn't notice that before.

Reply:
break is not missing here (at least it was not added intentionally). This (now common) code was not used in this path before. Let me check if states are right.

Reply:
Fixed